docs/source/extending/optimizer_engine.rst

```diff
@@ -0,0 +1,197 @@
+Optimizer
+=========
+The optimizer classes carry out the optimization of the workflow, the graph function. These
+classes usually contain the acronym **MCO**, but there is no particular reason why your optimizer
+should be a multi-criterion optimizer. In fact you could create optimizer classes that don't
+even optimize at all: the core requirement of these classes is that they return a point or points
+in :ref:`parameter space<optimization-ref>`, along with the associated criteria/KPIs/ojectives. These points *could be*
+minima or Pareto-effecient, but could just as well be any kind of sampling: e.g. a grid or
+random sample.
+
+We will illustrate how to design and use these classes, using the example of the
+Nevergrad optimizer that comes with the Nevergrad plugin. The source code for this
+optimizer can be examined
+`here <https://github.com/force-h2020/force-bdss-plugin-nevergrad/tree/master/force_nevergrad/mco>`_.
+
+
+``BaseMCOModel``
+----------------
+The TraitsUI to the optimizer. ::
+
+    class NevergradMCOModel(BaseMCOModel):
+
+        #: Algorithms available to work with
+        algorithms = Enum(
+            *NevergradMultiOptimizer.class_traits()["algorithms"].handler.values
+        )
+
+        #: Defines the allowed number of objective calls
+        budget = PositiveInt(100)
+
+        #: Display the generated points at runtime
+        verbose_run = Bool(True)
+
+        def default_traits_view(self):
+            return View(
+                Item("algorithms"),
+                Item("budget", label="Allowed number of objective calls"),
+                Item("verbose_run"),
+            )
+
+It exposes a set of optimizer parameters with an associated ``View``.
+
+.. figure:: images/optimizer_algo.png
+    :align: center
+    :scale: 70 %
+
+
+``BaseMCO``
+-----------
+Creates a ``BaseOptimizerEngine`` object and runs that engine on the workflow. ::
+
+    class NevergradMCO(BaseMCO):
+
+        def run(self, evaluator):
+            model = evaluator.mco_model
+
+            optim = NevergradMultiOptimizer(
+                algorithms=model.algorithms,
+                kpis=model.kpis,
+                budget=model.budget)
+
+            optimizer = AposterioriOptimizerEngine(
+                kpis=model.kpis,
+                parameters=model.parameters,
+                single_point_evaluator=evaluator,
+                verbose_run=model.verbose_run,
+                optimizer=optim
+            )
+
+            for index, (optimal_point, optimal_kpis) \
+                    in enumerate(optimizer.optimize()):
```

Comment:
Couple of name change suggestions (that we could also include in the source code)
```suggestion
            optimize = NevergradMultiOptimizer(
                algorithms=model.algorithms,
                kpis=model.kpis,
                budget=model.budget)

            engine = AposterioriOptimizerEngine(
                kpis=model.kpis,
                parameters=model.parameters,
                single_point_evaluator=evaluator,
                verbose_run=model.verbose_run,
                optimizer=optim
            )

            for index, (optimal_point, optimal_kpis) \
                    in enumerate(engine.optimize()):
```